compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -108,6 +127,13 @@ public class JBallerinaBackend extends CompilerBackend {
     private DiagnosticResult diagnosticResult;
     private boolean codeGenCompleted;
     private final List<JarConflict> conflictedJars;
+    private final Map<String, ByteArrayOutputStream> optimizedJarStreams = new HashMap<>();
```

Comment:
Shall we improve this to initialize it only if the --optimize flag is provided?

Reply:
Fixed in c4d7a21a506cf7d64b1532a60dfae97a75e6af5a. Made other fields used by the optimizer initialized conditionally as well.